arm-recoveryservices/2016-06-01/swagger/registeredidentities.json

```diff
@@ -0,0 +1,388 @@
+      },
+      "x-ms-discriminator-value": "AzureActiveDirectory"
+    },
+    "ResourceCertificateAndACSDetails": {
+      "description": "Certificate details representing the Vault credentials for ACS.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceCertificateDetails"
+        }
+      ],
+      "properties": {
+        "certificate": {
+          "description": "The base64 encoded certificate raw data string.",
+          "type": "string"
+        },
+        "resourceId": {
+          "format": "int64",
+          "description": "Vault resource id.",
+          "type": "integer"
+        },
+        "globalAcsNamespace": {
+          "description": "ACS namespace name - tenant for our service.",
+          "type": "string"
+        },
+        "globalAcsHostName": {
+          "description": "Acs mgmt host name to connect to.",
+          "type": "string"
+        },
+        "globalAcsRPRealm": {
+          "description": "Global ACS namespace RP realm.",
+          "type": "string"
+        },
+        "subject": {
+          "description": "Certificate Subject Name.",
+          "type": "string"
+        },
+        "validFrom": {
+          "format": "date-time",
+          "description": "Certificate Validity start Date time.",
+          "type": "string"
+        },
+        "validTo": {
+          "format": "date-time",
+          "description": "Certificate Validity End Date time.",
+          "type": "string"
+        },
+        "thumbprint": {
+          "description": "Certificate thumbrprint.",
+          "type": "string"
+        },
+        "friendlyName": {
+          "description": "Certificate friendlyname.",
+          "type": "string"
+        },
+        "issuer": {
+          "description": "Certificate issuer.",
+          "type": "string"
+        }
+      },
+      "x-ms-discriminator-value": "AccessControlService"
+    },
+    "ResourceCertificateDetails": {
+      "description": "Certificate details representing the Vault credentials.",
+      "type": "object",
+      "properties": {
+        "authType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorhpic chain of types.",
+          "type": "string"
+        }
+      },
+      "discriminator": "authType"
+    },
+    "ResourceCertificateDetailsResource": {
+      "description": "Certificate details representing the Vault credentials.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ResourceCertificateDetails"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionId": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "The subscription Id.",
+      "required": true,
+      "type": "string"
+    },
+    "ResourceGroupName": {
+      "name": "resourceGroupName",

```

Comment:
@dragonfly91 Our set of requirements is constantly in motion, "previous PRs" always let stuff pass that won't pass now, even if it's just because a reviewer missed something.
Regarding min/max length or regex patterns: If there is a well defined and strictly enforced (by your service) contract that is most likely *not* going to change soon, it makes sense to specify it here. For example: Thinking of `vaultName`, if your service rejects empty string, please do provide a min length of 1. The client side validation will not only fail faster but also provide a better exception!